x/leverage/keeper/oracle.go

```diff
@@ -119,6 +119,29 @@ func (k Keeper) TotalTokenValue(ctx sdk.Context, coins sdk.Coins, mode types.Pri
 	return total, nil
 }
 
+// WeightedBorrowValue returns the total value of all input tokens, each multiplied
+// by borrow factor (which is the minimum of 2.0 and 1/collateral weight). It
+// ignores unregistered and blacklisted tokens instead of returning an error, but
+// will error on unavailable prices.
+func (k Keeper) WeightedBorrowValue(ctx sdk.Context, coins sdk.Coins, mode types.PriceMode) (sdk.Dec, error) {
+	total := sdk.ZeroDec()
+
+	for _, c := range coins {
+		token, err := k.GetTokenSettings(ctx, c.Denom)
+		if err != nil {
+			continue
+		}
+		v, err := k.TokenValue(ctx, c, mode)
+		if err != nil {
```

Comment:
no - we cannot ignore missing prices of borrowed assets, becase that would allow users to borrow past their limit if a price was down for a token they had already borrowed